Bottom: fa1ed2fe4730e04e4c1d35e36541d0d7e8597d6d
Top:    d8daf49a74e643b633ddf7af67b5a1f2c233160c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:55:17 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6a3862f..7555599 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1160,7 +1160,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
